internal/api/dto/scheduled_task.go

```diff
@@ -0,0 +1,175 @@
+package dto
+
+import (
+	"time"
+
+	"github.com/flexprice/flexprice/internal/domain/scheduledtask"
+	ierr "github.com/flexprice/flexprice/internal/errors"
+	"github.com/flexprice/flexprice/internal/types"
+	"github.com/flexprice/flexprice/internal/validator"
+)
+
+// CreateScheduledTaskRequest represents a request to create a scheduled task
+type CreateScheduledTaskRequest struct {
+	ConnectionID string                 `json:"connection_id" binding:"required"`
+	EntityType   string                 `json:"entity_type" binding:"required"`
+	Interval     string                 `json:"interval" binding:"required"`
+	Enabled      bool                   `json:"enabled"`
+	JobConfig    map[string]interface{} `json:"job_config" binding:"required"`
+}
+
+// UpdateScheduledTaskRequest represents a request to update a scheduled task
+// Only enabled field is allowed to be updated
+type UpdateScheduledTaskRequest struct {
+	Enabled *bool `json:"enabled" validate:"required"`
+}
+
+// Validate validates the update request
+func (r *UpdateScheduledTaskRequest) Validate() error {
+	if r.Enabled == nil {
+		return ierr.NewError("enabled field is required").
+			WithHint("Please provide enabled field (true or false)").
+			Mark(ierr.ErrValidation)
+	}
+	return nil
+}
+
+// ScheduledTaskResponse represents a scheduled task response
+type ScheduledTaskResponse struct {
+	ID            string                 `json:"id"`
+	TenantID      string                 `json:"tenant_id"`
+	EnvironmentID string                 `json:"environment_id"`
+	ConnectionID  string                 `json:"connection_id"`
+	EntityType    string                 `json:"entity_type"`
+	Interval      string                 `json:"interval"`
+	Enabled       bool                   `json:"enabled"`
+	JobConfig     map[string]interface{} `json:"job_config"`
+	LastRunAt     *time.Time             `json:"last_run_at,omitempty"`
+	NextRunAt     *time.Time             `json:"next_run_at,omitempty"`
+	LastRunStatus string                 `json:"last_run_status,omitempty"`
+	LastRunError  string                 `json:"last_run_error,omitempty"`
+	Status        string                 `json:"status"`
+	CreatedAt     time.Time              `json:"created_at"`
+	UpdatedAt     time.Time              `json:"updated_at"`
+}
+
+// ListScheduledTasksResponse represents a list of scheduled tasks
+type ListScheduledTasksResponse struct {
```

Comment:
use the standard types
// ListInvoicesResponse represents the paginated response for listing invoices
type ListInvoicesResponse = types.ListResponse[*InvoiceResponse]